apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/GenesisImporter.java

```diff
@@ -14,122 +14,241 @@
+            final DatabaseManager databaseManager,
+            final AplAppStatus aplAppStatus,
+            final GenesisImporterProducer genesisImporterProducer,
+            final ApplicationJsonFactory jsonFactory,
+            final int publicKeyNumberTotal,
+            final int balanceNumberTotal
+    ) {
+        this.blockchainConfig = getBlockchainConfig(blockchainConfig);
+        this.blockchainConfigUpdater = getBlockchainConfigUpdater(blockchainConfigUpdater);
+        this.databaseManager = getDatabaseManager(databaseManager);
+        this.aplAppStatus = getAplAppStatus(aplAppStatus);
+        this.genesisParametersLocation = getGenesisParametersLocation(genesisImporterProducer);
+        this.jsonFactory = getJsonFactory(jsonFactory);
+        this.publicKeyNumberTotal = publicKeyNumberTotal;
+        this.balanceNumberTotal = balanceNumberTotal;
+    }
+
+    private ApplicationJsonFactory getJsonFactory(final ApplicationJsonFactory jsonFactory) {
+        return Objects.requireNonNull(jsonFactory, "jsonFactory is NULL");
+    }
+
+    private DatabaseManager getDatabaseManager(final DatabaseManager databaseManager) {
+        return Objects.requireNonNull(databaseManager, "databaseManager is NULL");
+    }
+
+    private String getGenesisParametersLocation(final GenesisImporterProducer genesisImporterProducer) {
+        return Optional.ofNullable(genesisImporterProducer)
+                .map(GenesisImporterProducer::genesisParametersLocation)
+                .orElseThrow(() -> new NullPointerException("genesisParametersLocation is NULL"));
+    }
+
+    private AplAppStatus getAplAppStatus(final AplAppStatus aplAppStatus) {
+        return Objects.requireNonNull(aplAppStatus, "aplAppStatus is NULL");
+    }
+
+    private BlockchainConfigUpdater getBlockchainConfigUpdater(
+            final BlockchainConfigUpdater blockchainConfigUpdater
+    ) {
+        return Objects.requireNonNull(blockchainConfigUpdater, "blockchainConfigUpdater is NULL");
+    }
+
+    private BlockchainConfig getBlockchainConfig(final BlockchainConfig blockchainConfig) {
+        return Objects.requireNonNull(blockchainConfig, "blockchainConfig is NULL");
     }
 
     @PostConstruct
     public void loadGenesisDataFromResources() {
         if (CREATOR_PUBLIC_KEY == null) {
-            try (InputStream is = ClassLoader.getSystemResourceAsStream(genesisParametersLocation)) {
-                JsonNode genesisParameters = mapper.readTree(is);
-                CREATOR_PUBLIC_KEY = Convert.parseHexString(genesisParameters.get("genesisPublicKey").asText());
-                CREATOR_ID = Account.getId(CREATOR_PUBLIC_KEY);
-                DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss Z");
-                EPOCH_BEGINNING = dateFormat.parse(genesisParameters.get("epochBeginning").asText()).getTime();
-             } catch (IOException | java.text.ParseException e) {
+            try (
+                    final InputStream is =
```

Comment:
Looks like JsonParser doesnt make a sense here, file is too small

Reply:
yes, but I decided to remove ObjectMapper completely in this class. It was initialized in a class member var without any CDI and required additional maintenance